### ISSUE TITLE
Question about the ground truth output

### ISSUE THREAD

**Issue by KingStorm** [lixilinx/IVA4Cocktail#1]
Hi, 

just came across your work, quite interesting of replacing laplace distribution with neural nets' density model. But I did not get crystal clear about: how do we get the ground truth output to train these NN density models?

**Comment by lixilinx** [comment]
Thanks for your interests.

This is not the popular end-to-end speech separation method, but is a well formulated optimization problem, and thus better generalizations to unseen configurations.

Basically, given the pdf of speech, then you estimate the maximum likelihood mixing or separation matrix, then you can get the separated results, then you can evaluate the goodness of your separations. This means that you can learn the pdf of speech to optimize the goodness (proxy objective) of the final separation results using gradient descent. In the training stage, the original source speeches are known. So you can evaluate how good is the separation given a pdf.
